plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -405,7 +405,21 @@ hoistExpr var t =
                     (PIR.Def var' (PIR.mkVar () var', PIR.Strict))
                     mempty
 
-                t' <- compileExpr t
+                CompileContext {ccOpts=profileOpts} <- ask
+                t' <- do
+                    if coProfile profileOpts==All then do
+                        let ty = PLC.varDeclType var'
+                        t'' <- compileExpr t
+                        let tInText = T.pack (show t'')
+                            -- Put "exiting" in the body
+                            traceInside (LamAbs () n x body) = LamAbs () n x (traceInside body)
+                            traceInside e                    = mkTrace ty ("exiting"<> tInText) e
```

Comment:
this is going to be the wrong type. If we have
```
f :: a -> b -> c
f = \a -> \b -> body
```
then `ty` is `a -> b -> c`, but you want to have `trace @c "exiting f" body` on the inside, so you need to strip off bits of the type too.

Reply:
```
traceInside :: Term -> Type -> Term
traceInside (LamAbs n t body) (TyFun dom cod) = LamAbs n t (traceInside body cod)
traceInside (LamAbs{}) _ = error "no"
traceInside e ty = mkTrace ty ...
```
What happens with functions like:
```
foo = \a -> \b -> ...
foo a b = 
```